sycl/plugins/unified_runtime/ur/adapters/hip/program.cpp

```diff
@@ -78,11 +80,14 @@ UR_APIEXPORT ur_result_t UR_APICALL
 urProgramCreateWithIL(ur_context_handle_t hContext, const void *pIL,
                       size_t length, const ur_program_properties_t *pProperties,
                       ur_program_handle_t *phProgram) {
-  ur_device_handle_t hDevice = hContext->getDevice();
-  const auto pBinary = reinterpret_cast<const uint8_t *>(pIL);
-
-  return urProgramCreateWithBinary(hContext, hDevice, length, pBinary,
-                                   pProperties, phProgram);
```

Comment:
Sorry just getting to this comment but will post PR in UR soon. Making this entry point unsupported since it doesn't take a device param. And since context to devices is now one to many mapping, we need something extra to tell us which device in the context this program refers to.